File: leetcode/C++/pairs-of-songs-with-total-durations-divisible-by-60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        int sol=0;
        vector<int> v(60,0);
        for(int &t : time) ++v[t%60];
        for(int i=0,j=60; i<=j; ++i,--j){
            if(i==0 || i==j) sol+=(v[i]*(v[i]-1))/2;
            else sol+=v[i]*v[j];
        }
        return sol;
    }
};

File: leetcode/Python/pairs-of-songs-with-total-durations-divisible-by-60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        sol, v, i, j = 0, [0]*60, 0 , 60
        for t in time:
            v[t%60]+=1
        while i<=j:
            if i==0 or i==j:
                sol+=(v[i]*(v[i]-1))//2
            else:
                sol+=v[i]*v[j]
            i, j = i+1, j-1
        return sol